oneview/resource_server_profile_template.go

```diff
@@ -221,32 +244,39 @@ func resourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{})
 	d.Set("mac_type", spt.MACType)
 	d.Set("hide_unused_flex_nics", spt.HideUnusedFlexNics)
 
-	networks := make([]map[string]interface{}, 0, len(spt.Connections))
-	for _, network := range spt.Connections {
-
-		networks = append(networks, map[string]interface{}{
-			"name":           network.Name,
-			"function_type":  network.FunctionType,
-			"network_uri":    network.NetworkURI,
-			"port_id":        network.PortID,
-			"requested_mbps": network.RequestedMbps,
-			"id":             network.ID,
-		})
+	var connections []ov.Connection
+	if len(spt.ConnectionSettings.Connections) != 0 {
+		connections = spt.ConnectionSettings.Connections
+	} else {
+		connections = spt.Connections
 	}
+	if len(connections) != 0 {
+		networks := make([]map[string]interface{}, 0, len(connections))
+		for _, network := range connections {
 
-	networkCount := d.Get("network.#").(int)
-	if networkCount > 0 {
-		for i := 0; i < networkCount; i++ {
-			currNetworkId := d.Get("network." + strconv.Itoa(i) + ".id")
-			for j := 0; j < len(spt.Connections); j++ {
-				if spt.Connections[j].ID == currNetworkId && i <= len(spt.Connections)-1 {
-					networks[i], networks[j] = networks[j], networks[i]
+			networks = append(networks, map[string]interface{}{
+				"name":           network.Name,
+				"function_type":  network.FunctionType,
+				"network_uri":    network.NetworkURI,
+				"port_id":        network.PortID,
+				"requested_mbps": network.RequestedMbps,
+				"id":             network.ID,
+			})
+		}
+		networkCount := len(connections)
+
+		if networkCount > 0 {
+			for i := 0; i < networkCount; i++ {
+				currNetworkId := d.Get("network." + strconv.Itoa(i) + ".id")
```

Comment:
We really need to `network` (and any other attribute that does a d.Get in a read function to fix the ordering) over to a TypeSet so we stop doing this crazy conversion.